create-or-update-version/createOrUpdateVersion.sh

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env bash
+
+MISSING=()
+[ ! "$PACT_BROKER_BASE_URL" ] && MISSING+=("PACT_BROKER_BASE_URL")
+[ ! "$PACT_BROKER_TOKEN" ] && MISSING+=("PACT_BROKER_TOKEN")
+[ ! "$version" ] && MISSING+=("version")
+[ ! "$application_name" ] && MISSING+=("application_name")
+
+if [ ${#MISSING[@]} -gt 0 ]; then
+  echo "ERROR: The following environment variables are not set:"
+  printf '\t%s\n' "${MISSING[@]}"
+  exit 1
+fi
+
+branch=$(git rev-parse --abbrev-ref HEAD)
+
+echo "
+PACT_BROKER_BASE_URL: '$PACT_BROKER_BASE_URL'
+PACT_BROKER_TOKEN: '$PACT_BROKER_TOKEN'
+version: '$version'
+application_name: '$application_name'
+branch: '$branch'
+"
+
+docker run --rm \
+    -e PACT_BROKER_BASE_URL=$PACT_BROKER_BASE_URL \
+    -e PACT_BROKER_TOKEN=$PACT_BROKER_TOKEN \
+    pactfoundation/pact-cli:latest \
+    broker create-or-update-version \
+    --pacticipant "$application_name" \
+    --version $version \
+    --branch $branch
```

Comment:
is branch required? if so it should be a workflow input i believe, but would need to check the others

Reply:
actually it should be --main-branch

https://github.com/pact-foundation/pact_broker-client#create-or-update-pacticipant

Reply:
It's `--branch`. You are looking at the other link. It should be https://github.com/pact-foundation/pact_broker-client#create-or-update-version

I tested by running this command on my own project and it look like `--branch` is not required

```shell
docker run --rm \                                                                                                                                                                                                                                                                              09:17:09
                               -e PACT_BROKER_BASE_URL=https://my-project.pactflow.io \
                               -e PACT_BROKER_TOKEN=MYtoken \
                               pactfoundation/pact-cli:latest \
                               broker create-or-update-version \
                               --pacticipant "name" \
                               --version 123
Created/updated pacticipant version 123 in PactFlow
```

Reply:
And yes, I'm not sure what is the best way to handle `--branch` there:
* Keep it like [publish-pact-files](https://github.com/pactflow/actions/blob/main/publish-pact-files/publishPactfiles.sh#L15)?
* Remove it completely?
* Convert it to input to allow user to change it?

Reply:
hmmm sorry my bad in reading the wrong docs, thanks for clarfication.

not sure if someone would want to create a pacticipant version with a branch every time? 

If yes, they setting it as per `publish-pact-files` might be good, so it is automatically determined and applied.

I wonder if a user may choose to only run this on certain branches (which they can stop via an if condition to run only if x branch) when executing the action.

maybe 
 - apply the branch by default as per [publish-pact-files](https://github.com/pactflow/actions/blob/main/publish-pact-files/publishPactfiles.sh#L15)
- allow user to override branch via branch input
- allow user to disable branch association (apply_branch=false, with the default value of true)

That might cover the cases where a user doesn't want to add a branch.

Reply:
So I think let keep it like [publish-pact-files](https://github.com/pactflow/actions/blob/main/publish-pact-files/publishPactfiles.sh#L15) for now

Reply:
yeah that makes sense to me :)